docs/Implementing-easyblocks.rst

```diff
@@ -254,6 +254,44 @@ redefining it entirely, since we call out to the original ``configure_step`` met
 Specific aspects of easyblocks
 ------------------------------
 
+.. _implementing_easyblocks_default_parameters:
+
+Default easyconfig parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All of the parameters which are "set" in an easyconfig file (see :ref:`vsd_avail_easyconfig_params`)
+become key-value pairs in the ``self.cfg`` dictionary.  For instance, if the easyconfig file specifies
+
+.. code:: python
+
+  name = 'example'
+  version = '2.5.3'
+  versionsuffix = '-Python-3.7.4'
+
+then these three parameters are accessible within an easyblock via
+
+.. code:: python
+
+  longform = ''.join(self.cfg['name'],'/',self.cfg['version'],self.cfg['versionsuffix'])
+
+You can use this notation successfully in your easyblock.  For all of the variables listed
+:ref:`vsd_avail_easyconfig_params`, the ``EasyBlock`` base class copies those dictionary entries into local Python
+variables:
+
+* ``self.name = self.cfg['name']``
+* ``self.version = self.cfg['version']``
+* ``self.versionsuffix = self.cfg['versionsuffix']``
```

Comment:
@kelseymh This is not correct actually...

`self.name` and `self.version` are special cases, see https://github.com/easybuilders/easybuild-framework/blob/d66c0e518d36a614e79c75fed73ec60d94b77f32/easybuild/framework/easyconfig/easyconfig.py#L1084-L1096 .

There is no `self.versionsuffix`, and in general you need to use `self.cfg['name_of_parameter']`.

Reply:
I'll get rid of self.versionsuffix, but it seems like there are other cases than just `self.name` and `self.version`.  Near the code you cited, I see `@property def toolchain(self)` and `@property def all_dependencies(self)`.

I continue to learn about Python.  The `@property` keyword is what allows these things to be reference without a function call (i.e., `self.name` instead of `self.name()`.  I'll update this to list just the four `@property` cases, and make it clear those are special.

Reply:
Updates are in PR #634